generic3g/tests/Test_Couplers.pf

```diff
@@ -0,0 +1,119 @@
+      _UNUSED_DUMMY(this)
+      
+   end subroutine setUp
+   
+   @After
+   subroutine shutDown(this)
+      class(ESMF_TestMethod), intent(inout) :: this
+
+      call ESMF_GeomDestroy(geom)
+      _UNUSED_DUMMY(this)
+
+   end subroutine shutDown
+
+   @Test(type=ESMF_TestMethod, npes=[1])
+   subroutine test_units(this)
+      class(ESMF_TestMethod), intent(inout) :: this
+      type(ESMF_StateItem_Flag), parameter :: ITEMTYPE = MAPL_STATEITEM_FIELD
+      character(len=*), parameter :: EXPORT_NAME = 'EXPORT_NAME'
+      character(len=*), parameter :: IMPORT_NAME = 'IMPORT_NAME'
+      type(ESMF_TypeKind_Flag), parameter :: TYPEKIND = ESMF_TYPEKIND_R4 
+      character(len=*), parameter :: EXPORT_UNITS = 'm s-1'
+      character(len=*), parameter :: IMPORT_UNITS = 'km s-1'
+      type(StateRegistry), target :: registry
+      type(StateRegistry), pointer :: regptr
+      type(VariableSpec) :: var_spec
+      type(StateItemSpec) :: export_spec, import_spec
+      type(VirtualConnectionPt) :: virtual_pt
+      type(StateItemExtension), pointer :: extension
+      type(StateItemSpec), pointer :: new_spec
+      class(StateItemAspect), pointer :: aspect
+      character(len=:), allocatable :: units
+      integer :: status
+
+      registry = StateRegistry('StateRegistry')
+      regptr => registry
+      @assertTrue(allocated(vertical_grid), 'The VerticalGrid has not been allocated.')
+      var_spec = make_VariableSpec(ESMF_STATEINTENT_EXPORT, EXPORT_NAME,&
+         & typekind=TYPEKIND, itemtype=ITEMTYPE, units=EXPORT_UNITS, _RC)
+      export_spec = var_spec%make_StateItemSpec(regptr, component_geom=geom,&
+         & vertical_grid=vertical_grid, _RC)
+      call export_spec%create(_RC)
+
+      var_spec = make_VariableSpec(ESMF_STATEINTENT_IMPORT, IMPORT_NAME,&
+         & typekind=TYPEKIND, itemtype=ITEMTYPE, units=IMPORT_UNITS, _RC)
+      import_spec = var_spec%make_StateItemSpec(regptr, component_geom=geom,&
+         & vertical_grid=vertical_grid, _RC)
+      call import_spec%create(_RC)
+
+      virtual_pt = VirtualConnectionPt(state_intent='export', short_name=EXPORT_NAME)
+!      call registry%add_virtual_pt(virtual_pt, _RC) 
+      call registry%add_primary_spec(virtual_pt=virtual_pt, spec=export_spec, _RC)
+      extension => registry%extend(virtual_pt, import_spec, _RC)
+      new_spec => extension%get_spec()
+      aspect => new_spec%get_aspect(UNITS_ASPECT_ID, _RC)
+      select type(aspect)
+      type is (UnitsAspect)
+         units = aspect%get_units()
+      end select
+      @assertEqual(IMPORT_UNITS, units)
+      _UNUSED_DUMMY(this)
+
+   end subroutine test_units
+
+!   export_spec = make_StateItem ...
+!   call export_spec%create(_RC)
+
+!   ! call to change metada - set units to 'm s-1'
+!   aspect => export_spec%get_aspect(GEOM_ASPECT_ID, _RC)
+!   select type (aspect)
+!   type UnitsAspect
+!     units = aspect%get_units()
+!  @assertEqual('m s-1', units)
+!   end  
```

Comment:
Can this be deleted at this point?  If you are going to follow up with additional tests than are based on this, I can approve the PR, and you can just continue.

```suggestion 
```